packages/wasm/crate/src/dex.rs

```diff
@@ -1,36 +1,37 @@
-use crate::utils;
-use penumbra_dex::lp::position::{Id, Position};
+use penumbra_dex::lp::position::{Id, Position, State};
 use penumbra_dex::lp::LpNft;
-use serde_wasm_bindgen::Error;
+use penumbra_proto::DomainType;
 use wasm_bindgen::prelude::wasm_bindgen;
-use wasm_bindgen::JsValue;
+
+use crate::error::WasmResult;
+use crate::utils;
 
 /// compute position id
 /// Arguments:
 ///     position: `Position`
 /// Returns: `PositionId`
 #[wasm_bindgen]
-pub fn compute_position_id(position: JsValue) -> Result<JsValue, Error> {
+pub fn compute_position_id(position: &[u8]) -> WasmResult<Vec<u8>> {
     utils::set_panic_hook();
 
-    let position: Position = serde_wasm_bindgen::from_value(position)?;
-    serde_wasm_bindgen::to_value(&position.id())
+    let position = Position::decode(position)?;
+    Ok(position.id().encode_to_vec())
 }
 
 /// get LP NFT asset
 /// Arguments:
 ///     position_value: `lp::position::Position`
 ///     position_state_value: `lp::position::State`
-/// Returns: `DenomMetadata`
+/// Returns: `Uint8Array representing a DenomMetadata`
 #[wasm_bindgen]
 pub fn get_lpnft_asset(
-    position_id_value: JsValue,
-    position_state_value: JsValue,
-) -> Result<JsValue, Error> {
+    position_id_value: &[u8],
+    position_state_value: &[u8],
```

Comment:
It makes sense to rename `position_id_value` -> `position_id` 
I added `_value` to mark `JsValue` parameters, but it didn't make sense either.